transcripts.py

```diff
@@ -191,47 +195,93 @@ def sort(transcript):
+	duration: typing.Optional[Interval] = None,
+	gap: typing.Optional[Interval] = None,
+	num_speakers: typing.Optional[Interval] = None,
+	allowed_audio_names: typing.Set[str] = None,
+	allowed_unk_count: typing.Optional[Interval] = None,
+	groups: typing.Set = None,
+	max_audio_file_size: typing.Optional[int] = None
 ):
+	if max_audio_file_size is not None and os.path.getsize(transcript[0]['audio_path']) > max_audio_file_size:
+		yield from ()
 	is_aligned = lambda w: (w.get('type') or w.get('error_tag')) == 'ok'
 	duration_check = lambda t: duration is None or duration[0] <= compute_duration(t) <= duration[1]
 	boundary_check = lambda t: ((not t.get('words')) or (not align_boundary_words) or
 								(is_aligned(t['words'][0]) and is_aligned(t['words'][-1])))
 	gap_check = lambda t, prev: prev is None or gap is None or gap[0] <= t['begin'] - prev['end'] <= gap[1]
-	unk_check = lambda t: unk is None or unk[0] <= t.get('ref', '').count('*') <= unk[1]
+	unk_check = lambda t: allowed_unk_count is None or allowed_unk_count[0] <= t.get('ref', '').count('*') <= allowed_unk_count[1]
 	speakers_check = lambda t: num_speakers is None or num_speakers[0] <= (t.get('speaker') or ''
 																			).count(',') + 1 <= num_speakers[1]
 	cer_check = lambda t: cer is None or t.get('cer') is None or cer[0] <= t['cer'] <= cer[1]
 	wer_check = lambda t: wer is None or t.get('wer') is None or wer[0] <= t['wer'] <= wer[1]
 	mer_check = lambda t: mer is None or t.get('mer') is None or mer[0] <= t['mer'] <= mer[1]
 	groups_check = lambda t: groups is None or t.get('group') is None or t['group'] in groups
+	name_check = lambda t: allowed_audio_names is None or audio_name(t) in allowed_audio_names
 
 	prev = None
 	for t in transcript:
 		if groups_check(t) and unk_check(t) and duration_check(t) and cer_check(t) and wer_check(t) and mer_check(
-			t) and boundary_check(t) and gap_check(t, prev) and speakers_check(t):
+			t) and boundary_check(t) and gap_check(t, prev) and speakers_check(t) and name_check(t):
 			yield t
 		prev = t
 
 
+def join_transcript(transcript: Transcript, join_channels: bool = False):
+	joined_transcripts = []
+
+	if join_channels:
+		groupped_t = [(channel_missing, transcript)]
+	else:
+		channel_key = lambda t: t.get('channel', channel_missing)
+		groupped_t = itertools.groupby(sorted(transcript, key = channel_key), channel_key)
+
+	for channel, transcript in groupped_t:
+		transcript = list(transcript)
+		audio_path = transcript[0]['audio_path']
+		assert all(t['audio_path'] == audio_path for t in transcript)
+		ref = ' '.join(t['ref'].strip() for t in transcript)
+		speaker = []
+		for t in transcript:
+			speaker.append(
+				torch.full((len(t['ref']) + 1,), fill_value = t['speaker'], dtype = torch.int64, device = 'cpu')
+					 .scatter_(0, torch.tensor(len(t['ref'])), speaker_missing) # space handling
+			)
+		speaker: shaping._T = torch.cat(speaker)[:-1].unsqueeze(0) # [:-1] to drop last space, because of len(t['ref'] + 1)
```

Comment:
As we discussed, you may move `unsqueeze` at a later point and use `_T` instead of `_T`